appservice/src/deploy/runWithZipStream.ts

```diff
@@ -39,13 +50,26 @@ export async function getZipStream(context: IActionContext, fsPath: string, clie
         } else {
             zipper.file(fsPath, { name: path.basename(fsPath) });
         }
-        await zipper.finalize();
         zipStream = zipper;
+        zipper.on('end', () => {
+            onFileSize(zipper.pointer());
```

Comment:
Can this not go in the generic task where end resolves the stream (L63)?

Reply:
Uh I could, but `pointer` is specific to `zipper` (aka it doesn't apply in the "alreadyZipped" case). I'd have to change the type of `zipStream` to
```
Readable & { finalize?(): Promise<void>; pointer?(): number };
```
And do a check like this:
```
if (zipStream.pointer) {
    onFileSize(zipStream.pointer());
}
```

I thought it was a little simpler to handle on `zipper` itself